subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidPluginsSmokeTest.groovy

```diff
@@ -192,6 +198,14 @@ class AndroidPluginsSmokeTest extends AbstractSmokeTest {
             expectNoDeprecationWarnings(result)
         }
 
+        when:
+        writeActivity(library, libPackage, libraryActivity, true)
+        result = runner('build', '-x', 'lint').build()
+
+        then:
+        result.task(':library:compileReleaseJavaWithJavac').outcome == TaskOutcome.SUCCESS
+        result.task(':app:compileReleaseJavaWithJavac').outcome == TaskOutcome.SUCCESS
+
```

Comment:
It would be nice to somehow document that this is about incremental build. Add a comment? Move it to a separate test method?

Reply:
I added a name for the `when`/`then` methods. Do you think that is enough?